### PR TITLE
Crowd analysis

### PR DESCRIPTION
**Pull request by mojtaba-komeili**
**Patch description**
1. A bit of refactoring on the crowdsourcing analysis module
2. Added the filtering function to help discarding low quality unit

### CODE REVIEW

File: parlai/crowdsourcing/utils/analysis.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ def setup_args(cls):
         )
         return parser
 
-    def __init__(self, opt: Dict[str, Any]):
+    def __init__(self, opt: Opt):
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/utils/analysis.py:51]: 🎉

File: parlai/crowdsourcing/utils/analysis.py [hunk 2]
```diff
@@ -61,6 +62,19 @@ def get_results_path_base(self) -> str:
             f'{self.__class__.__name__}__{now.strftime("%Y%m%d_%H%M%S")}',
         )
 
+    def unit_acceptable(self, unit_data: Dict[str, Any]) -> bool:
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/utils/analysis.py:65]: nit: `is_unit_acceptable` might be clearer

File: parlai/crowdsourcing/utils/analysis.py [hunk 3]
```diff
@@ -185,18 +199,26 @@ def get_task_units(self, task_name: str) -> List[Unit]:
         data_browser = self.get_mephisto_data_browser()
         return data_browser.get_units_for_task_name(task_name)
 
-    def get_units_data(self, task_units: List[Unit]) -> List[dict]:
+    def get_data_from_unit(self, unit: Unit) -> Dict[str, Any]:
+        """
+        Retrieves task data for a single unit.
+        """
+        try:
+            data_browser = self.get_mephisto_data_browser()
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/utils/analysis.py:207]: hmm how expensive is the operation of getting the Mephisto data browser? Will doing it once for every unit lead to a slowdown?

**Review by EricMichaelSmith — COMMENTED** [review]
Great to have! Minor nits/questions

File: parlai/crowdsourcing/utils/analysis.py (hunk 3 above)
Comment (mojtaba-komeili) [reply parlai/crowdsourcing/utils/analysis.py:207]: It is not using Mephisto one directly.  It uses [this](https://github.com/facebookresearch/ParlAI/blob/e062f5d58b88f3e843de8d6635c5aa8018b6a66b/parlai/crowdsourcing/utils/analysis.py#L177) method that keeps the browser object reference, the first time that it accessed it.

Comment (EricMichaelSmith) [reply parlai/crowdsourcing/utils/analysis.py:207]: oh ha great :)

**Review by EricMichaelSmith — APPROVED** [review]

**Pull request merged**